src/order/ideal.lean

```diff
@@ -50,6 +60,17 @@ structure ideal (P) [preorder P] :=
 (directed  : directed_on (≤) carrier)
 (mem_of_le : ∀ {x y : P}, x ≤ y → y ∈ carrier → x ∈ carrier)
 
+/-- A predicate for when a subset of `P` is an ideal. -/
+@[mk_iff] structure is_ideal {P} [preorder P] (I : set P) : Prop :=
+(nonempty : I.nonempty)
+(directed : directed_on (≤) I)
+(mem_of_le : ∀ {x y : P}, x ≤ y → y ∈ I → x ∈ I)
```

Comment:
Could we actually spell out the definition in the doc-string? I admit I know nothing about ideals in preorders, but I'm a little sad that I have to rely on mouse hover or jump to definition on `directed_on` before I can even know what one is.

Reply:
(The doc-string for `ideal` would be fine too. I just want the word `directed` explained.)

Reply:
@bryangingechen, [this change](https://github.com/leanprover-community/mathlib/pull/6656/commits/1ae3f643c1f3e6d0584bfe80f7f70829b93c61d9) didn't actually fix my concern. I want to know what `directed` means in the first place.

(I appreciate that this is not very mathlib-y, to actually give informal definitions when structures are being introduced, but perhaps it should be. :-)

Reply:
Oh, sorry about that. I didn't read your message carefully enough.